### PR TITLE
 feat(http): add http write client 

### LINKED ISSUE #1025: Add HTTP write client

### LINKED ISSUE #1026: Add writing data to CLI
Follows #1025

### PR DESCRIPTION
**Pull request by goller**
Closes #1025 
Closes #1026 

_Briefly describe your proposed changes:_
Add an HTTP write client and write to the CLI program.
_What was the problem?_
We had neither
_What was the solution?_
added client, tests and mock of points writer.

  - [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergeable
  - [x] Tests pass
  - [x] swagger.json updated (if modified Go structs or API)
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: cmd/influx/write.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	buckets, _, err := bs.FindBuckets(ctx, filter)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	bucketID, orgID := buckets[0].ID, buckets[0].OrganizationID
+
+	var r io.Reader
+	if args[0] == "-" {
```
Comment (fntlnz) [inline cmd/influx/write.go]: Check args len here before accessing

File: mock/points_writer.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	p.Points = append(p.Points, points...)
+	err := p.Err
+	p.mu.Unlock()
+	return err
+}
+
+// Next returns the next (oldest) batch of values.
+func (p *PointsWriter) Next() models.Point {
+	var points models.Point
+	p.mu.RLock()
```
Comment (fntlnz) [inline mock/points_writer.go]: I'm missing something here or the unlock can just be deffered here and be valid for the whole function?

File: cmd/influx/write.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	"strings"
+
+	"github.com/influxdata/platform"
+	"github.com/influxdata/platform/http"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var writeCmd = &cobra.Command{
+	Use:   "write [line protocol or @/path/to/points.txt",
```
Comment (leodido) [inline cmd/influx/write.go]: Typo

File: mock/points_writer.go (hunk 2 above)
Comment (fntlnz) [reply mock/points_writer.go]: right, thanks for clarifying

File: cmd/influx/write.go (hunk 1 above)
Comment (goller) [reply cmd/influx/write.go]: @fntlnz args are required by cobra here: https://github.com/influxdata/platform/pull/1072/files#diff-a1dc84d7004d75c4e674f016b31ff0ceR21

```go
Args: cobra.ExactArgs(1),
```

File: write/batcher.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+	maxBytes := b.MaxFlushBytes
+	if maxBytes == 0 {
+		maxBytes = DefaultMaxBytes
+	}
+
+	buf := make([]byte, 0, maxBytes)
+	var line []byte
+	var more = true
+
```
Comment (e-dard) [inline write/batcher.go]: You could add `r := bytes.NewReader(buf)` here.

File: write/batcher.go [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+	// if read closes the channel normally, exit the loop
+	for more {
+		select {
+		case line, more = <-lines:
+			if more {
+				buf = append(buf, line...)
+			}
+			// write if we exceed the max lines OR read routine has finished
+			if len(buf) >= maxBytes || (!more && len(buf) > 0) {
+				r := bytes.NewReader(buf)
```
Comment (e-dard) [inline write/batcher.go]: Then here, instead of allocating a new reader for each chunk you can do `r.Reset(buf)`.

**Pull request merged**